src/StorageReader/ClaimController.ts

```diff
@@ -39,41 +39,31 @@ export interface Arguments {
   readonly configuration: ClaimControllerConfiguration
 }
 
-export class ClaimController {
-  private readonly logger: Pino.Logger
-  private readonly db: Db
-  private readonly collection: Collection
-  private readonly messaging: Messaging
-  private readonly ipfs: IPFS
-  private readonly configuration: ClaimControllerConfiguration
-  private readonly verifiableClaimSigner: VerifiableClaimSigner
-
-  constructor({
-    dependencies: {
-      logger,
-      db,
-      messaging,
-      ipfs,
-      verifiableClaimSigner,
-    },
-    configuration,
-  }: Arguments) {
-    this.logger = childWithFileName(logger, __filename)
-    this.db = db
-    this.collection = this.db.collection('storageReader')
-    this.configuration = configuration
-    this.messaging = messaging
-    this.ipfs = ipfs
-    this.verifiableClaimSigner = verifiableClaimSigner
-  }
+export interface ClaimController {
+  readonly download: (ipfsFileHashes: ReadonlyArray<string>) => Promise<void>
+  readonly downloadNextHash: (options?: { retryDelay?: number, maxAttempts?: number }) => Promise<void>
+}
 
-  async download(ipfsFileHashes: ReadonlyArray<string>) {
-    const logger = this.logger.child({ method: 'download' })
+export const ClaimController = ({
+  dependencies: {
+    logger,
+    db,
+    messaging,
+    ipfs,
+    verifiableClaimSigner,
+  },
+  configuration,
+}: Arguments): ClaimController => {
+  const childLogger = childWithFileName(logger, __filename)
```

Comment:
How about this instead?
```suggestion
  const claimControllerLogger = childWithFileName(logger, __filename)
```

Reply:
Thanks, that does work better, I will update.